structuralcodes/sections/_generic.py

```diff
@@ -0,0 +1,461 @@
+        '''Increases the global counter by one'''
+        cls.section_counter += 1
+
+    @classmethod
+    def return_global_counter_and_increase(cls):
+        '''Returns the current counter and increases it by one'''
+        counter = cls.section_counter
+        cls._increase_global_counter()
+        return counter
+
+
+class PointGeometry(Geometry):
+    '''Class for a point geometry with material.
+
+    Basically it is a wrapper for shapely Point
+    including the material (and other parameters that
+    may be needed)'''
+    def __init__(
+        self,
+        point: t.Union[Point, ArrayLike],
+        diameter: float,
+        material: Material,
+        name: t.Optional[str] = None
+    ):
+        super().__init__(name)
+        # I check if point is a shapely Point or an ArrayLike object
+        if not isinstance(point, Point):
+            # It is an ArrayLike object -> create the Point given the
+            # coordinates (coordinates can be a List, Tuple, np.array, ...)
+            coords = np.atleast_1d(point)
+            num = len(coords)
+            if num < 2:
+                raise ValueError('Two coordinates are needed')
+            if num > 2:
+                warn_str = f'Two coordinates are needed. {num}'
+                warn_str += ' coords provided. The extra entries will be'
+                warn_str += ' discarded'
+                warnings.warn(warn_str)
+            point = Point(coords)
+        if not isinstance(material, Material):
+            raise TypeError('The material should be of Material class')
+        self._point = point
+        self._diameter = diameter
+        self._material = material
+        self._area = np.pi * diameter**2 / 4.0
+
+    @property
+    def diameter(self) -> float:
+        '''Returns the point diameter'''
+        return self._diameter
+
+    @property
+    def area(self) -> float:
+        '''Returns the point area'''
+        return self._area
+
+    @property
+    def material(self) -> Material:
+        '''Returns the point material'''
+        return self._material
+
+    @property
+    def x(self) -> float:
+        '''Returns the x coordinate of the point'''
+        return self._point.x
+
+    @property
+    def y(self) -> float:
+        '''Returns the y coordinate of the point'''
+        return self._point.y
+
+    @property
+    def point(self) -> Point:
+        '''Returns the shapely Point object'''
+        return self._point
+
+
+def create_line_point_angle(
+    point: t.Union[Point, t.Tuple[float, float]],
+    theta: float,
+    bbox: t.Tuple[float, float, float, float]
+) -> LineString:
+    '''Creates a line from point and angle within the bounding
+    box'''
+    # create a unit vector defining the line
+    v = (np.cos(theta), np.sin(theta))
+
+    # check if the line is vertical to avoid div by zero
+    if abs(v[0]) > 1e-8:
+        # it is a non vertical line
+        tg = v[1] / v[0]
+        x1 = bbox[0] - 1e-3
```

Comment:
Why are we subtracting and adding the small number to the bounds of the bounding box? I just don't get it 😃